python/ic_azure/insights_metric.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/python
+
+# Python imports
+from datetime import datetime, timedelta
+from argparse import ArgumentParser
+
+# Azure client imports
+from azure_client import AzureClient
+
+
+class InsightsMetric(object):
+    """Retrieve metric data from Azure's Microsoft Insights components."""
+
+    def __init__(self, azure_client):
+        self._client = azure_client.client()
+        self.subscription_id = azure_client.subscription_id
+
+    # Method to retrieve metrics from Microsoft Insight resources
+    def get_metric(self, resource_group, resource_name, interval, metric,
+                   statistic, timeshift):
+
+        # Declare variables
+        ret_val = -1
+
+        # Calculate timestamps
+        end_time = datetime.utcnow() - timedelta(seconds=timeshift)
+        start_time = end_time - timedelta(seconds=interval)
+
+        # Create resource ID
+        resource_id = "subscriptions/{}/resourceGroups/{}".format(
+            self.subscription_id,
+            resource_group
+        )
+        resource_id += "/providers/microsoft.insights/components/{}".format(
+            resource_name
+        )
+
+        # Get timegrain based on interval
+        # Possible values: None, PT1M, PT1H, P1D
+        quotient, remainder = divmod(interval, 60)
+        if quotient <= 1:
+            timegrain = 'PT1M'
+        elif quotient <= 60:
```

Comment:
So if I give interval of 5 minutes the script actually gets PT1H metric ie. one-hour metric? Start time is however end time - 5 minutes; I wonder if the query will return the latest one-hour metric or is it possible that it does return nothing?

This is probably ok but I am thinking if the script could just take PT1M, PT1H or P1D as an argument instead of the interval. Would it be more explicit to use and understand what the script does?

Reply:
This is true. I will make the timegrain as an argument.

Reply:
This was changed in commit 895de0926cb3071bac7c208f93ee3668d39a30d4.

Reply:
What if interval was calculated automatically based on the timegrain argument. Eg for PT1M the interval would be [now - timeshift - 1 minute .... now - timeshift] ?

Reply:
Excellent idea. I will change this too.

Reply:
This was changed in commit 31fd1a2a9d163efb4fc2c0648b77558a7798f16c.